byteps/torch/parallel/distributed.py

```diff
@@ -180,6 +181,29 @@ def __init__(self, module, device_ids=None,
         if len(module_states) > 0:
             bps.torch.broadcast_parameters(self.module.state_dict(), root_rank=0)
 
+    @contextmanager
+    def no_sync(self):
+        r"""
+        A context manager to disable gradient synchronizations across DDP
+        processes. Within this context, gradients will be accumulated on module
+        variables, which will later be synchronized in the first
+        forward-backward pass exiting the context.
+
+        Example::
+
+            >>> ddp = byteps.torch.parallel.DistributedDataParallel(model, pg)
+            >>> with ddp.no_sync():
+            ...   for input in inputs:
+            ...     ddp(input).backward()  # no synchronization, accumulate grads
+            ... ddp(another_input).backward()  # synchronize grads
+        """
```

Comment:
could you add a check like this? https://github.com/bytedance/byteps/blob/master/byteps/torch/__init__.py#L178-L179

Reply:
added.